File: src/main/kotlin/lottoDraw/domain/vo/LottoDraw.kt
package lottoDraw.domain.vo

import lotto.domain.vo.LottoNumber
import lotto.domain.vo.LottoNumbers
import java.time.LocalDate

data class LottoDraw(
    val round: Int,
    val date: LocalDate,
    val numbers: LottoNumbers,
    val bonus: LottoNumber
) {
    fun matchResult(numbers: LottoNumbers): LottoResult {
        val count = numberCheck(numbers)
        val bonus = bonusCheck(numbers)

        return LottoResult(
            round = round,
            matchCount = count,
            bonusMatch = bonus
        )
    }

    private fun bonusCheck(numbers: LottoNumbers): Boolean = numbers.contains(bonus)
    private fun numberCheck(numbers: LottoNumbers): Int = this.numbers.check(numbers)
}

File: frontend/types/lotto-draw.ts
import { List } from "postcss/lib/list";

export interface LottoDrawPreviewNumberResponse {
    success: boolean;
    status: number;
    message: string;
    data: LottoDrawPreviewResponse
} 
export interface LottoNumberCountResponse{
    number: number;
    count: number;    
}
export interface LottoDrawPreviewResponse{
    lottoNumberCountResponse: LottoNumberCountResponse[],
    lottoResultResponse: LottoResultResponse[]
}
 
export interface LottoResultResponse{
    round: number;
    lottoDrawResponse : LottoDrawResponse,
    drawResultResponse : DrawResultResponse
}

export interface LottoDrawResponse{
    date: Date,
    numbers: number[],
    bonus: number
}
export interface DrawResultResponse{
    matchCount: number,
    bonusMatch: boolean
}
